--- pycrypto-2.3/setup.py	2010-08-27 09:25:59.462963720 +0100
+++ pycrypto-2.3/setup.py	2010-08-27 10:12:04.590901603 +0100
@@ -136,7 +136,7 @@
             self.compiler.include_dirs.insert(0, "src/inc-msvc/")
 
         # Detect libgmp and don't build _fastmath if it is missing.
-        lib_dirs = self.compiler.library_dirs + ['/lib', '/usr/lib']
+        lib_dirs = self.compiler.library_dirs + ['/lib64', '/usr/lib64']
         if not (self.compiler.find_library_file(lib_dirs, 'gmp')):
             print >>sys.stderr, "warning: GMP library not found; Not building Crypto.PublicKey._fastmath."
             self.__remove_extensions(["Crypto.PublicKey._fastmath"])
