--- setup.py.old	2004-12-18 12:52:44.031165688 +0100
+++ setup.py	2004-12-18 12:54:16.861053400 +0100
@@ -116,7 +116,7 @@
         build_ext.build_extensions(self)
 
     def detect_modules (self):
-        lib_dirs = self.compiler.library_dirs + ['/lib', '/usr/lib']
+        lib_dirs = self.compiler.library_dirs + ['/lib64', '/usr/lib64']
         inc_dirs = self.compiler.include_dirs + ['/usr/include']
         exts = []
         if (self.compiler.find_library_file(lib_dirs, 'gmp')):
